dask/dataframe/partitionquantiles.py

```diff
@@ -414,11 +420,21 @@ def percentiles_summary(df, num_old, num_new, upsample, state):
     if is_categorical_dtype(data):
         data = data.cat.codes
         interpolation = "nearest"
-    elif isinstance(data.dtype, pd.core.dtypes.dtypes.DatetimeTZDtype) or np.issubdtype(
-        data.dtype, np.integer
-    ):
+    elif is_datetime64tz_dtype(data.dtype) or is_integer_dtype(data.dtype):
         interpolation = "nearest"
-    vals, n = _percentile(data, qs, interpolation=interpolation)
+
+    # FIXME: pandas quantile doesn't work with some data types (e.g. strings).
+    # We fall back to an ndarray as a workaround.
+    if ArrowNotImplementedError:
+        # `string[pyarrow]` dtypes will throw `ArrowNotImplementedError`
+        exceptions = (TypeError, ArrowNotImplementedError)
+    else:
+        exceptions = (TypeError,)
+    try:
+        vals = data.quantile(q=qs / 100, interpolation=interpolation)
```

Comment:
```suggestion
        vals = data.quantile(q=qs / 100, interpolation=interpolation).values
```

Not certain, but I *think* this addresses the cudf/cupy errorr we discussed offline. My uderstanding is that we are getting a cudf object here when we want an array.

Reply:
Thanks for the suggestion @rjzamora -- just pushed a commit to try this suggestion out!

Reply:
gpuCI is now happy 🎉